coderd/externalauth/externalauth.go

```diff
@@ -811,6 +819,24 @@ func (c *jwtConfig) Exchange(ctx context.Context, code string, opts ...oauth2.Au
 	)
 }
 
+// When authenticating via Entra ID ADO only supports v1 tokens that requires the 'resource' rather than scopes
+// When ADO gets support for V2 Entra ID tokens this struct and functions can be removed
+type jwtConfigEntraV1 struct {
+	*oauth2.Config
+}
+
+func (c *jwtConfigEntraV1) AuthCodeURL(state string, opts ...oauth2.AuthCodeOption) string {
+	return c.Config.AuthCodeURL(state, append(opts, oauth2.SetAuthURLParam("resource", "499b84ac-1321-427f-aa17-267ca6975798"))...)
```

Comment:
Can we make `499b84ac-1321-427f-aa17-267ca6975798` a `const` and use that in both the `AuthCodeURL` and `Exchange` functions? 

Then comment what this magic hash is. At first glance, a random static hash looks incorrect, but it's some constant on Microsoft's side.

Reply:
Created a const called `azureDevOpsAppID`